experimental/FTheoryTools/src/types.jl

```diff
@@ -0,0 +1,107 @@
+  ambient_space::ToricOrNonToricCoveredScheme
+  defining_ideal::MPolyIdeal
+  # zero section
+  CompleteIntersectionModel(base_space::ToricOrNonToricCoveredScheme, ambient_space::ToricOrNonToricCoveredScheme, defining_ideal::MPolyIdeal) = new(base_space, ambient_space, defining_ideal)
+end
+
+
+@attributes mutable struct HypersurfaceModel <: AbstractFTheoryModel
+  base_space::ToricOrNonToricCoveredScheme
+  ambient_space::ToricOrNonToricCoveredScheme
+  hypersurface_equation::MPolyRingElem{QQFieldElem}
+  HypersurfaceModel(base_space::ToricOrNonToricCoveredScheme, ambient_space::ToricOrNonToricCoveredScheme, hypersurface_equation::MPolyRingElem{QQFieldElem}) = new(base_space, ambient_space, hypersurface_equation)
+end
+
+
+@attributes mutable struct GlobalWeierstrassModel
+  weierstrass_f::MPolyRingElem{QQFieldElem}
+  weierstrass_g::MPolyRingElem{QQFieldElem}
+  weierstrass_polynomial::MPolyRingElem{QQFieldElem}
+  base_space::ToricOrNonToricCoveredScheme
+  ambient_space::ToricOrNonToricCoveredScheme
+  fiber_ambient_space::ToricOrNonToricCoveredScheme
+  function GlobalWeierstrassModel(weierstrass_f::MPolyRingElem{QQFieldElem},
+                            weierstrass_g::MPolyRingElem{QQFieldElem},
+                            weierstrass_polynomial::MPolyRingElem{QQFieldElem},
+                            base_space::ToricOrNonToricCoveredScheme,
+                            ambient_space::ToricOrNonToricCoveredScheme)
+    return new(weierstrass_f, weierstrass_g, weierstrass_polynomial, base_space, ambient_space, weighted_projective_space(ToricCoveredScheme, [2,3,1]))
+  end
+end
+
+
+@attributes mutable struct GlobalTateModel
+  tate_a1::MPolyRingElem{QQFieldElem}
+  tate_a2::MPolyRingElem{QQFieldElem}
+  tate_a3::MPolyRingElem{QQFieldElem}
+  tate_a4::MPolyRingElem{QQFieldElem}
+  tate_a6::MPolyRingElem{QQFieldElem}
+  tate_polynomial::MPolyRingElem{QQFieldElem}
+  base_space::ToricOrNonToricCoveredScheme
+  ambient_space::ToricOrNonToricCoveredScheme
+  fiber_ambient_space::ToricOrNonToricCoveredScheme
+  function GlobalTateModel(tate_a1::MPolyRingElem{QQFieldElem},
+                          tate_a2::MPolyRingElem{QQFieldElem},
+                          tate_a3::MPolyRingElem{QQFieldElem},
+                          tate_a4::MPolyRingElem{QQFieldElem},
+                          tate_a6::MPolyRingElem{QQFieldElem},
+                          tate_polynomial::MPolyRingElem{QQFieldElem},
+                          base_space::ToricOrNonToricCoveredScheme,
+                          ambient_space::ToricOrNonToricCoveredScheme)
+    return new(tate_a1, tate_a2, tate_a3, tate_a4, tate_a6, tate_polynomial, base_space, ambient_space, weighted_projective_space(ToricCoveredScheme, [2,3,1]))
+  end
+end
+
+
+################################################
+# 2: conceptual parents
+################################################
+
+@attr ClosedSubschemeModel function conceptual_parent(cim::CompleteIntersectionModel)
```

Comment:
This looks interesting. But I do not understand it straight away; I would like to have these functions explained to me later.